spring-ai-model/src/main/java/org/springframework/ai/tool/execution/DefaultToolCallResultConverter.java

```diff
@@ -45,7 +47,11 @@ public final class DefaultToolCallResultConverter implements ToolCallResultConve
 	public String convert(@Nullable Object result, @Nullable Type returnType) {
 		if (returnType == Void.TYPE) {
 			logger.debug("The tool has no return type. Converting to conventional response.");
-			return JsonParser.toJson("Done");
+			return "Done";
+		}
+		if (returnType instanceof Class<?> cls && cls.equals(String.class)) {
+			logger.debug("Tool return type is String. Returning result as is.");
+			return Optional.ofNullable((String) result).orElse("null");
```

Comment:
What about more simple variant?
```suggestion
			return String.valueOf(result);
```
This will return `"null"` for null `result` and string representation for non-null `result`, same as suggested by PR but more efficiently.

Reply:
Thanks for the feedback.
This is implemented intentionally to maintain backward compatibility with version 1.0.0. The previous JsonParser.toJson(result) method returned the literal string "null" for a null input, and this code preserves that exact behavior.

Reply:
Thank you for pointing that out.
You’re absolutely right that String.valueOf(result) exactly matches the behavior I intended: it returns the literal string "null" when result is null, without needing an Optional.